CoreFoundation/Base.subproj/CFPlatform.c

```diff
@@ -884,11 +884,15 @@ CF_EXPORT int _NS_rename(const char *oldName, const char *newName) {
 }
 
 CF_EXPORT int _NS_open(const char *name, int oflag, int pmode) {
+#if TARGET_OS_WIN32
+    return _CFOpenFileWithMode(name, oflag, pmode);
+#else
     wchar_t *wide = createWideFileSystemRepresentation(name, NULL);
     int fd;
     _wsopen_s(&fd, wide, oflag, _SH_DENYNO, _S_IREAD | _S_IWRITE);
     free(wide);
     return fd;
+#endif
```

Comment:
This code is Windows specific already, so this change doesn't make sense.  Just kill the existing implementation.  Although, I think that we want to inline the definition that you placed in `_CFOpenFileWithMode` here as `open` is being replaced with `_NS_open` on Windows.